tests/iterator.rs

```diff
@@ -67,3 +67,10 @@ fn signals_block_wait() {
         .expect_err("Wait didn't wait properly");
     assert_eq!(err, RecvTimeoutError::Timeout);
 }
+
+#[cfg(not(any(target_os = "linux", windows)))]
+use signal_hook::SIGINFO;
+#[test]
```

Comment:
Is this test iterator related? I'd say this belongs somewhere slightly differently than tests/iterator.

Style:
* I prefer to have all uses together at the top.
* There should be a space above `#[test]` to visually separate the two items.

Also note that the CI fails on linux because of this, you omit importing the name, but you don't omit the test using it itself.